src/state/farms/elasticv2/hooks.ts

```diff
@@ -0,0 +1,258 @@
+import { gql, useLazyQuery } from '@apollo/client'
+import { defaultAbiCoder } from '@ethersproject/abi'
+import { getCreate2Address } from '@ethersproject/address'
+import { keccak256 } from '@ethersproject/solidity'
+import { CurrencyAmount, Token, WETH } from '@kyberswap/ks-sdk-core'
+import { FeeAmount, Pool, Position } from '@kyberswap/ks-sdk-elastic'
+import { BigNumber } from 'ethers'
+import { Interface } from 'ethers/lib/utils'
+import { useEffect } from 'react'
+
+import FarmV2QuoterABI from 'constants/abis/farmv2Quoter.json'
+import NFTPositionManagerABI from 'constants/abis/v2/ProAmmNFTPositionManager.json'
+import { NETWORKS_INFO } from 'constants/networks'
+import { EVMNetworkInfo } from 'constants/networks/type'
+import { NativeCurrencies } from 'constants/tokens'
+import { useActiveWeb3React } from 'hooks'
+import { useContract, useMulticallContract } from 'hooks/useContract'
+import { useKyberswapConfig } from 'hooks/useKyberswapConfig'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { isAddressString } from 'utils'
+
+import { defaultChainData, setFarms, setLoading, setUserFarmInfo } from '.'
+import { ElasticFarmV2, SubgraphFarmV2, SubgraphToken, UserFarmV2Info } from './types'
+
+const queryFarms = gql`
+  {
+    farmV2S {
+      id
+      startTime
+      endTime
+      pool {
+        id
+        feeTier
+        tick
+        sqrtPrice
+        liquidity
+        reinvestL
+        token0 {
+          id
+          symbol
+          name
+          decimals
+        }
+        token1 {
+          id
+          symbol
+          name
+          decimals
+        }
+      }
+      rewards(orderBy: index, orderDirection: asc) {
+        token {
+          id
+          symbol
+          name
+          decimals
+        }
+        amount
+        index
+      }
+      ranges {
+        id
+        index
+        isRemoved
+        tickUpper
+        tickLower
+        weight
+      }
+    }
+  }
+`
+
+let isSubscribed = false
```

Comment:
move inside component and make it a useRef?

Reply:
no, I want this is global, not only access by one component.
for example, when 2 components use this hook, it's only setInterval 1 time